### PR TITLE
add new piecewise linear transform functions

### PR DESCRIPTION
**Pull request by fxlim25**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: digital_image_processing/intensity_transformation/piecewise.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    transformation that is used for image enhancement. We obtain low contrast image 
+    due to poor illumination (wrong setting of lens aperture), and that effect can 
+    be overcome using contrast stretching. The basic idea behind contrast stretching 
+    is to increase the contrast of an image by making darker portion more darker and 
+    brighter portion brighter.
+    """
+    
+    if (0 <= img and img <= r1):
+        return (s1/r1)*img
+    elif (r1 <= img and img <= r2):
```
Comment (cclauss) [inline digital_image_processing/intensity_transformation/piecewise.py:24]:
```suggestion
    if 0 <= img <= r1:
        return (s1/r1)*img
    elif r1 <= img <= r2:
```

**Comment by cclauss** [comment]
How does the test prove that this algorithm is working?

**Comment by fxlim25** [comment]
> How does the test prove that this algorithm is working?

@cclauss i run the code it seems like give desired result i want, do you have idea on how to fix the test code?

**Comment by cclauss** [comment]
Click `détails` to the right of ❌

**Pull request closed without merging**